packages/odyssey-react/src/components/Box/Box.tsx

```diff
@@ -11,128 +11,501 @@
+    | "help"
+    | "progress"
+    | "wait"
+    | "cell"
+    | "crosshair"
+    | "text"
+    | "vertical-text"
+    | "alias"
+    | "copy"
+    | "move"
+    | "no-drop"
+    | "not-allowed"
+    | "all-scroll"
+    | "col-resize"
+    | "row-resize"
+    | "n-resize"
+    | "e-resize"
+    | "s-resize"
+    | "w-resize"
+    | "ns-resize"
+    | "nw-resize"
+    | "se-resize"
+    | "sw-resize"
+    | "nesw-resize"
+    | "nwse-resize";
+  /**
+   * Mouse interaction
+   */
+  pointerEvents?: "none" | "auto";
   /**
    * Content selection
    */
-  userSelect?: "none" | "auto";
+  userSelect?: "none" | "text" | "contain" | "all" | "auto";
 }
 
 function convertProp(prop: string) {
-  return prop.charAt(0).toUpperCase() + prop.substring(1).toLowerCase();
+  function clearAndUpper(text: string) {
+    return text.replace(/-/, "").toUpperCase();
+  }
+  return prop.replace(/(^\w|-\w)/g, clearAndUpper);
 }
 
 /**
  * Low level building block for ui
  */
-let Box = forwardRef<HTMLDivElement, BoxProps>((props, ref) => {
-  const {
-    borderColor,
-    hoverBorderColor,
-    borderRadius,
-    boxShadow,
-    hoverBoxShadow,
-    display,
-    flexGrow,
-    alignItems,
-    focusRing,
-    marginRight,
-    marginBottom,
-    padding,
-    cursor,
-    userSelect,
-    children,
-    ...rest
-  } = props;
+let Box = forwardRef(
+  (
+    {
+      as: Tag = "div",
+      display,
+      position,
+      flexDirection,
+      flexWrap,
+      flex,
+      flexGrow,
+      flexShrink,
+      alignItems,
+      justifyContent,
+      gridTemplateColumns,
+      gridColumn,
+      gridRow,
+      width,
+      maxWidth,
+      height,
+      maxHeight,
+      margin,
+      marginTop,
+      marginRight,
+      marginBottom,
+      marginLeft,
+      padding,
+      paddingTop,
+      paddingRight,
+      paddingBottom,
+      paddingLeft,
+      overflow,
+      overflowX,
+      overflowY,
+      borderColor,
+      hoverBorderColor,
+      focusBorderColor,
+      borderRadius,
+      borderTopLeftRadius,
+      borderTopRightRadius,
+      borderBottomLeftRadius,
+      borderBottomRightRadius,
+      backgroundColor,
+      boxShadow,
+      hoverBoxShadow,
+      focusRing,
+      cursor,
+      pointerEvents,
+      userSelect,
+      children,
+      ...rest
+    },
+    ref
+  ) => {
+    if (Array.isArray(margin)) {
+      switch (margin.length) {
+        case 2:
+          marginTop = margin[0];
+          marginBottom = margin[0];
+          marginLeft = margin[1];
+          marginRight = margin[1];
+          break;
+        case 3:
+          marginTop = margin[0];
+          marginRight = margin[1];
+          marginLeft = margin[1];
+          marginBottom = margin[2];
+          break;
+        case 4:
+          marginTop = margin[0];
+          marginRight = margin[1];
+          marginBottom = margin[2];
+          marginLeft = margin[3];
+          break;
+        default:
+          margin = margin[0];
+      }
+    }
```

Comment:
can we break this logic apart into separate pure functions with their own tests even? it will help keep the code modular and easier to maintain I hope